src/smart-components/Channel/context/hooks/useHandleChannelEvents.ts

```diff
@@ -0,0 +1,208 @@
+import { useEffect } from "react";
+import { GroupChannel, GroupChannelHandler, SendbirdGroupChat } from "@sendbird/chat/groupChannel";
+import { FileMessage, UserMessage } from "@sendbird/chat/message";
+
+import { scrollIntoLast } from '../utils';
+
+import { CustomUseReducerDispatcher, Logger } from "../../../../lib/SendbirdState";
+import uuidv4 from "../../../../utils/uuid";
+import compareIds from '../../../../utils/compareIds';
+import * as messageActions from '../dux/actionTypes';
+
+/**
+ * Handles ChannelEvents and send values to dispatcher using messagesDispatcher
+ * messagesDispatcher: Dispatcher
+ * sdk: sdkInstance
+ * logger: loggerInstance
+ * channelUrl: string
+ * sdkInit: bool
+ */
+interface DynamicParams {
+  sdkInit: boolean;
+  hasMoreNext: boolean;
+  currentGroupChannel: GroupChannel;
+}
+interface StaticParams {
+  sdk: SendbirdGroupChat;
+  logger: Logger;
+  scrollRef: React.RefObject<HTMLDivElement>;
+  setQuoteMessage: React.Dispatch<React.SetStateAction<UserMessage | FileMessage>>;
+  messagesDispatcher: CustomUseReducerDispatcher;
+}
+
+function useHandleChannelEvents(
+  {
+    sdkInit,
+    hasMoreNext,
+    currentGroupChannel,
+  }: DynamicParams,
+  {
+    sdk,
+    logger,
+    scrollRef,
+    setQuoteMessage,
+    messagesDispatcher,
+  }: StaticParams,
+): void {
+  useEffect(() => {
+    const channelUrl = currentGroupChannel?.url;
+    const channelHandlerId = uuidv4();
+    if (channelUrl && sdkInit) {
+      const channelHandler = {
+        onMessageReceived: (channel, message) => {
+          // Do not update when hasMoreNext
+          if (compareIds(channel?.url, channelUrl) && !hasMoreNext) {
+            let scrollToEnd = false;
+            try {
+              const { current } = scrollRef;
+              scrollToEnd = current.offsetHeight + current.scrollTop >= current.scrollHeight;
+            } catch (error) {
+              //
+            }
+
+            logger.info('Channel | useHandleChannelEvents: onMessageReceived', message);
+            messagesDispatcher({
+              type: messageActions.ON_MESSAGE_RECEIVED,
+              payload: { channel, message },
+            });
```

Comment:
Hello, we don't have an issue tab, so please understand that we are asking about the PR. 

Is there a reason why it is not updated when the hasMoreNext value is true? I am facing an issue. 
Normally, the `messageActions.ON_MESSAGE_RECEIVED` event is executed when a message is received. 
However, there is a problem that the `messageActions.ON_MESSAGE_RECEIVED` event does not run if you move to the chat room even once,  So, `allMessages` array does not contain newly received messages.

> if you move to the chat room even once

I looked at the differences in the conditions, the `hasMoreNext` value was confirmed to be true. 

I'd really appreciate it if you could take a look at the issue.

thanks

Reply:
Actually, I'm sorry that I took the liberty of coming to github because I reported it to the community but didn't get the answer I wanted. I'll make PR with easier-to-understand comments. Thank you for your reply 🙇

Reply:
@sravan-s Hi! I'll attach a pr with the relevant code.

PR link: https://github.com/sendbird/sendbird-uikit-react/pull/301